### ISSUE TITLE
Use with Babel?

### ISSUE THREAD

**Issue by Vaccano** [frankwallis/plugin-typescript#108]
Is it possible to use this with Babel?

What I am hoping for is that I can use this to compile from Typescript to ES6.  (So I can use Async/Await in my Typescript), then have Babel compile to ES5?

If this is possible, what kind of config would it take?

**Comment by masaeedu** [comment]
Why was this closed?

**Comment by Vaccano** [comment]
@masaeedu I closed it because I realized that the plugin is a runtime thing and Babel is a compile time thing. 

What I wanted was out of order.

**Comment by frederikschubert** [comment]
But you can use the typescript - es6 - es5 pipeline with this plugin. Just have a look at the examples/angular2 folder in this repo. You set the typescript loader for your project files and the Babel transpiler that takes the es6 and transpiles it to es5.

**Comment by masaeedu** [comment]
@frederikschubert I've already tried using that, but either babel transpilation isn't occurring, or it isn't detecting the configured presets. I have:

```js
System.config({
// ...
  babelOptions: {
    "blacklist": [],
    "stage": 0,
    "optional": [
      "runtime",
      "optimisation.modules.system"
    ]
  },
// ...
});
```

But when I attempt to compile:

```
var [x, y] = [1, 2];
```

I get:

> Error: ReferenceError: Invalid left-hand side in assignment

That either means no transpilation is taking place, or that the `babelOptions` setting is ignored.

**Comment by frankwallis** [comment]
I would expect this to work so it is some kind of configuration issue - it looks like you are using jspm 0.16 and babel 5, in which case your config should look something like this:

https://github.com/frankwallis/plugin-typescript/blob/72bd341708ec7cac9546a4628f277674cf674b4f/example/angular2/config.js

As @frederikschubert said the example angular2 project uses the full typescript -> babel pipeline and shows async/await working. I have recently upgraded it to jspm 0.17.0-beta / babel6 though, so to see it working with jspm 0.16.0 / babel 5 you would need to checkout commit 72bd341.

**Comment by masaeedu** [comment]
@frankwallis Thanks, that clarifies what the problem is somewhat. Does babel 6 only work with JSPM beta?

**Comment by frankwallis** [comment]
for babel 6 you need to use [plugin-babel](https://github.com/systemjs/plugin-babel) rather than the built in babel transpiler. I think it did work with jspm 0.16 as well as 0.17 but I am not certain.

**Comment by lastmjs** [comment]
@masaeedu Did you get this working? I haven't gotten into it yet, and I'd like to know if there are any secrets to getting it to work. Thanks!

**Comment by masaeedu** [comment]
@lastmjs No, sorry. I didn't get it to work. FWIW I just use `tsc -watch` now: output ES6 from the TypeScript compiler, set JSPM to transpile using babel and point it at the tsc output directory. Probably works okay for moderately sized codebases.

**Comment by lastmjs** [comment]
@frederikschubert I'm really trying to get this to work, can you perhaps post an example configuration object? And thanks @masaeedu, I'll see what I can do. I'm trying to get completely away from a build step.

**Comment by lastmjs** [comment]
@frederikschubert @masaeedu I think I've got it working. Here is the magic configuration:

```
    System.config({
                transpiler: 'plugin-babel',
                typescriptOptions: {
                    target: 'es6',
                    module: 'es6'
                },
                meta: {
                    '*.ts': {
                        loader: 'ts'
                    }
                },
                packages: {
                    ts: {
                        main: 'plugin.js'
                    },
                    typescript: {
                        main: 'typescript.js',
                        meta: {
                            'typescript.js': {
                                exports: 'ts'
                            }
                        }
                    }
                },
                map: {
                    ts: 'node_modules/plugin-typescript/lib/',
                    typescript: 'node_modules/typescript/lib/',
                    'plugin-babel': 'node_modules/systemjs-plugin-babel/plugin-babel.js',
                    'systemjs-babel-build': 'node_modules/systemjs-plugin-babel/systemjs-babel-browser.js'
                }
            });
```

**Comment by lastmjs** [comment]
And I wrote a library, just import it along with SystemJS: https://github.com/lastmjs/es-no-build
